kernel/src/process/posix_thread/futex.rs

```diff
@@ -540,16 +535,19 @@ impl FutexKey {
             );
         }
 
-        Ok(Self { addr, bitset, pid })
-    }
+        // Use `jhash` to hash `addr` and `pid`.
+        let hash = {
+            let addr_low = addr as u32;
+            let addr_high = (addr >> 32) as u32;
+            let pid_adjusted = pid.unwrap_or(u32::MAX);
+            jhash::jhash_2vals(addr_low, addr_high, pid_adjusted) as usize
```

Comment:
I suggest adding the following comment to explain the significance of setting the right _seed_ for jhash.

```rust
        let hash = {
            let addr_low = addr as u32;
            let addr_high = (addr >> 32) as u32;
            // Choose a different jhash seed (or salt) for each process (unless the futex is shared)
            // to prevent common key patterns from causing excessive collisions in the table.
            let seed = pid.unwrap_or(u32::MAX);
            jhash::jhash_2vals(addr_low, addr_high, seed) as usize
        }
```